### PR TITLE
Improve logging system and error handling

### PR DESCRIPTION
**Pull request by aarontorres0**
- Implement enhanced logging system to capture both INFO and ERROR levels effectively (Issue #3).
- Refine error traceback handling using logging for clearer, more actionable error messages (Issue #5).

### CODE REVIEW

File: src/DiscordBot.py [hunk 1]
```diff
@@ -10,23 +10,33 @@
...
-from dotenv import load_dotenv
-from datetime import datetime
-import traceback
 
 load_dotenv()
 DISCORD_TOKEN = os.getenv("DISCORD_TOKEN")
 CHANNEL_ID = os.getenv("CHANNEL_ID")
 GITHUB_TOKEN = os.getenv("GIT_TOKEN")
 
+logging.basicConfig(
```
Comment (DavidSalazar123) [inline src/DiscordBot.py:28]: Can you please add some comments here about the set up

File: src/DiscordBot.py [hunk 2]
```diff
@@ -35,83 +45,84 @@ async def scheduled_task(
...
 
-            print("All jobs have been posted!")
+            logging.info("All jobs have been posted!")
         else:
-            print("No new jobs! Time: ", datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
-
-    except Exception as e:
-        traceback.print_exc()
+            logging.info(
+                f"No new jobs! Time: {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}"
```
Comment (DavidSalazar123) [inline src/DiscordBot.py:83]: Please include the current_date here. We are more interested in what the time was when the task started

File: src/DiscordBot.py [hunk 3]
```diff
@@ -35,83 +45,84 @@ async def scheduled_task(
...
 ):
     """
-    A scheduled task that runs every 60 seconds to check for new commits in the GitHub repository
-
-    Parameters:
-        - github_utilities: An instance of the GitHubUtilities class
-        - internship_utilities: An instance of the InternshipUtilities class
+    A scheduled task that runs every 60 seconds to check for new commits in the GitHub repository.
     """
+    start_time = datetime.now()
```
Comment (DavidSalazar123) [inline src/DiscordBot.py:50]: Change `start_time` to `current_date` and other places as well

File: src/InternshipUtilities.py [hunk 4]
```diff
@@ -29,34 +30,34 @@ def __init__(self, summer: bool, coop: bool):
 
     def clearJobLinks(self) -> None:
         """
-        Clear the Co-Op dictionary links
+        Clear the Co-Op dictionary links.
```
Comment (DavidSalazar123) [inline src/InternshipUtilities.py:33]: Please remove all the periods

File: .gitignore [hunk 5]
```diff
@@ -8,3 +8,5 @@ __pycache__/
 
 # Environment variables
 .env
+
+repository_links_commits.json
```
Comment (DavidSalazar123) [inline .gitignore:12]: Forgot to include `commits/`

Comment (aarontorres0) [reply .gitignore:12]: Good catch

File: src/DiscordBot.py (hunk 1 above)
Comment (aarontorres0) [reply src/DiscordBot.py:28]: Yessir

File: src/DiscordBot.py (hunk 3 above)
Comment (aarontorres0) [reply src/DiscordBot.py:50]: Will keep `start_time`

File: src/DiscordBot.py (hunk 2 above)
Comment (aarontorres0) [reply src/DiscordBot.py:83]: Will use `start_time`

File: src/InternshipUtilities.py (hunk 4 above)
Comment (aarontorres0) [reply src/InternshipUtilities.py:33]: Ruff put these in

**Review by DavidSalazar123 — APPROVED** [review]
LGTM

**Pull request merged**